pylint/extensions/magic_value.py

```diff
@@ -0,0 +1,89 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Checks for magic values instead of literals."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker, utils
+from pylint.checkers.base.comparison_checker import ComparisonChecker
+from pylint.interfaces import HIGH
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class MagicValueChecker(BaseChecker):
+    """Checks for constants in comparisons."""
+
+    name = "magic-value"
+    msgs = {
+        "R2004": (
+            "Consider using a named constant or an enum instead of '%s'.",
+            "magic-value-compare",
```

Comment:
I think Pierre already suggested something here, but should we consider `magic-value-comparison`? It sounds a bit nicer imo.